src/identities/masterpubkey.ts

```diff
@@ -0,0 +1,342 @@
+import { AddressInterface } from '../types';
+import Identity, {
+  IdentityInterface,
+  IdentityType,
+  IdentityOpts,
+} from '../identity';
+import { BufferMap, isValidXpub, isValidBlindPub, toXpub } from '../utils';
+import { BIP32Interface, fromBase58 } from 'bip32';
+import { Slip77Interface, fromMasterBlindingKey } from 'slip77';
+import { payments } from 'liquidjs-lib';
+
+export interface MasterPublicKeyOptsValue {
+  masterPublicKey: string;
+  masterBlindingKey: string;
+}
+
+function instanceOfMasterPublicKeyOptsValue(
+  value: any
+): value is MasterPublicKeyOptsValue {
+  return 'masterPublicKey' in value && 'masterBlindingKey' in value;
+}
+
+interface AddressInterfaceExtended {
+  address: AddressInterface;
+  publicKey: string;
+  derivationPath: string;
+}
+
+// util function that parse a derivation path and return the index
+function getIndex(addrExtended: AddressInterfaceExtended) {
+  const derivationPathSplitted = addrExtended.derivationPath.split('/');
+  const index: number = parseInt(
+    derivationPathSplitted[derivationPathSplitted.length - 1]
+  );
+
+  return index;
+}
+
+export class MasterPublicKey extends Identity implements IdentityInterface {
+  static INITIAL_BASE_PATH: string = "m/84'/0'/0'";
```

Comment:
This should be: 
```suggestion
  static INITIAL_BASE_PATH: string = "m/84'/0'";
```

Do not hardcode the `account` of the derivation path.

Reply:
This has been copied from the Mnemonic implementation. At the moment we assume that the user CANNOT decide the account, we always default to 0.

If we do not hardocde there, where?

Reply:
Ideally, it should not be hardcoded at all.
The identity should have an incremental address index for every account rather than just one. It should be the user the one who specifies the account of the derivation path, but if you say that also other interfaces have this behavior, then it's ok for me as you've done but we should open a ticket to support multiple accounts.